### PR TITLE
feat: [SRM-11311]: NewRelic MS template

### PR DESCRIPTION
**Pull request by deepesh-ui**
### Summary

NewRelic Monitored Service template

#### Screenshots
Screenshot 2022-08-03 at 10 11 53 PM
Screenshot 2022-08-03 at 10 12 02 PM
Screenshot 2022-08-03 at 10 12 18 PM

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 17 Issues 
* Code issue 17 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10372)

File: src/modules/85-cv/components/InputWithDynamicModalForJson/InputWithDynamicModalForJson.tsx [hunk 1]
```diff
@@ -91,29 +103,72 @@ export function InputWithDynamicModalForJson(props: InputWithDynamicModalForJson
...
-            </Button>
+            {content}
           </Layout.Vertical>
         )
       }}
     />
   )
 }
+
+export const InputWithDynamicModalForJsonMultiType = (
```
Comment (praneshg239) [inline src/modules/85-cv/components/InputWithDynamicModalForJson/InputWithDynamicModalForJson.tsx:139]: Shall we extract this new component to new file?

Comment (deepesh-ui) [reply src/modules/85-cv/components/InputWithDynamicModalForJson/InputWithDynamicModalForJson.tsx:140]: InputWithDynamicModalForJsonMultiType is a MultiType version of existing question , have wrapped the component with already existing <ExpressionAndRuntimeType />

File: src/modules/85-cv/pages/health-source/connectors/NewRelic/NewRelicHealthSource.tsx [hunk 2]
```diff
@@ -154,7 +173,8 @@ export default function NewRelicHealthSource({
     if (
       newRelicData.isEdit &&
       selectedMetricPacks.length &&
```
Comment (harshilbits) [inline src/modules/85-cv/pages/health-source/connectors/NewRelic/NewRelicHealthSource.tsx:175]: Entire logic in if condition can be moved to utils ?

File: src/modules/85-cv/pages/health-source/connectors/NewRelic/NewRelicHealthSource.utils.tsx [hunk 3]
```diff
@@ -200,6 +204,19 @@ export const createNewRelicFormData = (
...
 
+export const setApplicationIfConnectorIsInput = (
+  isConnectorRuntimeOrExpression: boolean,
+  nonCustomFeilds: any,
+  setNonCustomFeilds: (data: any) => void
+): void => {
+  if (isConnectorRuntimeOrExpression) {
+    setNonCustomFeilds({
+      ...nonCustomFeilds,
+      newRelicApplication: '<+input>'
```
Comment (harshilbits) [inline src/modules/85-cv/pages/health-source/connectors/NewRelic/NewRelicHealthSource.utils.tsx:215]: use RUN_TIME_VALUE

File: src/modules/85-cv/pages/health-source/connectors/NewRelic/components/NewRelicCustomMetricForm/NewRelicCustomMetricForm.tsx [hunk 4]
```diff
@@ -99,9 +111,14 @@ export default function NewRelicCustomMetricForm(props: NewRelicCustomFormInterf
   }, [queryParamsForNRQL])
 
   useEffect(() => {
-    if (query?.trim().length) {
+    if (
+      query?.trim().length &&
```
Comment (harshilbits) [inline src/modules/85-cv/pages/health-source/connectors/NewRelic/components/NewRelicCustomMetricForm/NewRelicCustomMetricForm.tsx:115]: Please move this logic to a utils function with a meaningful name.

Comment (deepesh-ui) [reply src/modules/85-cv/pages/health-source/connectors/NewRelic/components/NewRelicCustomMetricForm/NewRelicCustomMetricForm.tsx]: done

File: src/modules/85-cv/pages/health-source/connectors/NewRelic/NewRelicHealthSource.utils.tsx (hunk 3 above)
Comment (deepesh-ui) [reply src/modules/85-cv/pages/health-source/connectors/NewRelic/NewRelicHealthSource.utils.tsx]: done

File: src/modules/85-cv/pages/health-source/connectors/NewRelic/NewRelicHealthSource.tsx (hunk 2 above)
Comment (deepesh-ui) [reply src/modules/85-cv/pages/health-source/connectors/NewRelic/NewRelicHealthSource.tsx]: done

File: src/modules/85-cv/components/InputWithDynamicModalForJson/InputWithDynamicModalForJson.tsx (hunk 1 above)
Comment (deepesh-ui) [reply src/modules/85-cv/components/InputWithDynamicModalForJson/InputWithDynamicModalForJson.tsx]: done

**Review by praneshg239 — APPROVED** [review]

**Comment by deepesh-ui** [comment]
retrigger jest

**Review by praneshg239 — APPROVED** [review]

**Review by vasuki-kalluri — APPROVED** [review]

**Review by harshilbits — APPROVED** [review]

**Pull request merged**